BSseq_pipeline.py

```diff
@@ -585,14 +595,40 @@
         methSegGR        = os.path.join(WORKDIR,DIR_seg,"{prefix}.sorted_meth_segments_gr.RDS"),
         methSegBed      = os.path.join(WORKDIR,DIR_seg,"{prefix}.sorted_meth_segments.bed"),
         methSegPng         = os.path.join(WORKDIR,DIR_seg,"{prefix}.sorted_meth_segments.png"),
-        methylDiff_files      = lambda wc: [ os.path.join(WORKDIR,rds) for rds in finalReportDiffMeth_input(wc.prefix)],
         genome_dir  = config['locations']['genome-dir'],
         scripts_dir = DIR_scripts,
-        CpGfile     = config['general']['differential-methylation']['annotation']['CpGfile'],
         refGenes_bedfile  = config['general']['differential-methylation']['annotation']['refGenes_bedfile'],
         webfetch    = config['general']['differential-methylation']['annotation']['webfetch']
     log:
         os.path.join(DIR_final,"{prefix}.sorted_{assembly}_final.log")
     message: fmt("Compiling final report.")
     run:
         generateReport(input, output, params, log, "")
+
+
+## Final Report for diff. meth.
+rule diffmeth_report:
+    input:
+        lambda wc: DIR_diffmeth + str(wc.treatment).replace('vs', '_') + '.sorted_diffmeth.bed',
+        template      = os.path.join(DIR_templates,"diffmeth.Rmd"),
+        chrom_seqlengths    = os.path.join(DIR_mapped,"Refgen_"+ASSEMBLY+"_chromlengths.csv")
+    output:
+        report        = os.path.join(DIR_final, "diffmeth-report.{treatment}.html")
+    params:
+        chrom_seqlengths  = os.path.join(DIR_mapped,"Refgen_"+ASSEMBLY+"_chromlengths.csv"),
+        source_dir  = config['locations']['input-dir'],
+        out_dir     = config['locations']['output-dir'],
+        diffmeth_dir = DIR_diffmeth,
+        treatment = lambda wc: str(wc.treatment).replace('vs', '_'),
+        assembly    = ASSEMBLY,
+        genome_dir  = config['locations']['genome-dir'],
+        scripts_dir = DIR_scripts,
+        CpGfile     = config['general']['differential-methylation']['annotation']['CpGfile'],
```

Comment:
Could we change the name of the cpg-islands bed file from "CpGfile", "cpgIsland_bedfile"?

Reply:
I agree. More descriptive names are generally desireable